src/box/lua/net_box.lua

```diff
@@ -1201,8 +1249,20 @@ index_metatable = function(remote)
         if opts and opts.buffer then
             error("index:count() doesn't support `buffer` argument")
         end
-        local code = string.format('box.space.%s.index.%s:count',
-                                   self.space.name, self.name)
+        local space_qualifier
+        if self.space.name then
+            space_qualifier = "." .. self.space.name
+        else
+            space_qualifier = "[" .. self.space.id .. "]"
+        end
+        local idx_qualifier
+        if self.name then
+            idx_qualifier = "." .. self.name
+        else
+            idx_qualifier = "[" .. self.id .. "]"
+        end
+        local code = string.format('box.space%s.index%s:count',
+                                   space_qualifier, idx_qualifier)
```

Comment:
Nit: "qualifier" doesn't seem to be a right word choice for this variable IMO. Anyway, using both string formatting and concatenation looks ugly. Let's use only concatenation since we need it anyway:

```Lua
        local code = 'box.space' .. (self.space.name ~= nil and
                                     '.' .. self.space.name or
                                     '[' .. self.space.id .. ']') ..
                     'index' .. (self.name ~= nil and
                                 '.' .. self.name or '[' .. self.id .. ']') ..
                     ':count'
```

Reply:
Applied your suggestion, thanks!